content/en/docs/ops/diagnostic-tools/virtual-machines/index.md

```diff
@@ -0,0 +1,121 @@
+
+The Istio sidecar is typically run as a `systemd` unit. To ensure its running properly, you can check that status:
+
+{{< text bash >}}
+$ systemctl status istio
+{{< /text  >}}
+
+Additionally, the sidecar health can be programmatically check at its health endpoint:
+
+{{< text bash >}}
+$ curl localhost:15021/healthz/ready -I
+{{< /text  >}}
+
+## Logs
+
+Logs for the Istio proxy can be found in a few places.
+
+To access the `systemd` logs, which has details about the initialization of the proxy:
+
+{{< text bash >}}
+$ journalctl -f -u istio -n 1000
+{{< /text  >}}
+
+The proxy will redirect `stderr` and `stdout` to `/var/log/istio/istio.err.log` and  `/var/log/istio/istio.log`, respectively.
+To view these in a format similar to `kubectl`:
+
+{{< text bash >}}
+$ tail /var/log/istio/istio.err.log /var/log/istio/istio.log -Fq -n 100
+{{< /text  >}}
+
+Log levels can be modified by changing the `cluster.env` configuration file. Make sure to restart `istio` if it is already running:
+
+{{< text bash >}}
+$ echo "ISTIO_AGENT_FLAGS=\"--log_output_level=dns:debug --proxyLogLevel=debug\"" >> /var/lib/istio/envoy/cluster.env
+$ systemctl restart istio
+{{< /text  >}}
+
+## Iptables
+
+To ensure `iptables` rules have been successfully applied:
+
+{{< text bash >}}
+$ sudo iptables-save
+...
+-A ISTIO_OUTPUT -d 127.0.0.1/32 -j RETURN
+-A ISTIO_OUTPUT -j ISTIO_REDIRECT
+{{< /text  >}}
+
+## Istioctl
+
+Most `istioctl` commands will function properly with virtual machines. For example, `istioctl proxy-status` can be used to view all connected proxies:
+
+{{< text bash >}}
+$ istioctl proxy-status
+NAME           CDS        LDS        EDS        RDS      ISTIOD                    VERSION
+vm-1.default   SYNCED     SYNCED     SYNCED     SYNCED   istiod-789ffff8-f2fkt     {{< istio_full_version >}}
+{{< /text  >}}
+
+However, `istioctl proxy-config` relies on functionality in Kubernetes to connect to a proxy, which will not work for virtual machines.
+Instead, a file containing the configuration dump from Envoy can be passed. For example:
+
+{{< text bash >}}
+$ curl -s localhost:15000/config_dump | istioctl proxy-config clusters --file -
+SERVICE FQDN                            PORT      SUBSET  DIRECTION     TYPE
+istiod.istio-system.svc.cluster.local   443       -       outbound      EDS
+istiod.istio-system.svc.cluster.local   15010     -       outbound      EDS
+istiod.istio-system.svc.cluster.local   15012     -       outbound      EDS
+istiod.istio-system.svc.cluster.local   15014     -       outbound      EDS
+{{< /text  >}}
+
+## Automatic registration
+
+When a virtual machine connects to Istiod, a `WorkloadEntry` will automatically be created. This enables
+the virtual machine to become a part of a `Service`, similar to an `Endpoint` in Kubernetes.
+
+To check these are created correctly:
+
+{{< text bash >}}
+$ kubectl get workloadentries
+NAME             AGE   ADDRESS
+vm-10.128.0.50   14m   10.128.0.50
+{{< /text  >}}
+
```

Comment:
I think we should add some details on troubleshooting workload entries not created automatically. grepping the istiod logs may be good enough for now.

Reply:
SGTM, I will add a note to look at logs. I also filed a bug to add metrics which can be helpful